From f2192280faacd22b8a27ca9877bb1a9f5112664c Mon Sep 17 00:00:00 2001
From: Manuel Rota <manuel.rota@kalpa.it>
Date: Mon, 22 Jul 2024 15:26:10 +0000
Subject: [PATCH] feat: add dual imx219 configuration

---
 imx/run.sh | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/imx/run.sh b/imx/run.sh
index b8ca8c4..9298b0b 100755
--- a/imx/run.sh
+++ b/imx/run.sh
@@ -35,6 +35,7 @@ USAGE+="\tdual_os08a20_1080p30hdr - dual os08a20 cameras on MIPI-CSI1/2, 1920x10
 USAGE+="\tos08a20_4khdr           - single os08a20 camera on MIPI-CSI1, 3840x2160, 15 fps, HDR configuration\n"
 
 USAGE+="\timx219_1080p60         - single imx219 camera on MIPI-CSI1, 1920x1080, 60 fps\n"
+USAGE+="\tdual_imx219_1080p60     - dual imx219 camera on MIPI-CSI1/2, 1920x1080, 60 fps\n"
 
 # parse command line arguments
 while [ "$1" != "" ]; do
@@ -328,7 +329,16 @@ case "$ISP_CONFIG" in
                          MODE="0"
                          write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
                          ;;
-
+                 dual_imx219_1080p60 )
+                         MODULES=("imx219" "${MODULES[@]}")
+                         RUN_OPTION="DUAL_CAMERA"
+                         CAM_NAME="imx219"
+                         DRV_FILE="imx219.drv"
+                         MODE_FILE="IMX219_MODES.txt"
+                         MODE="0"
+                         write_sensor_cfg_file "Sensor0_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
+                         write_sensor_cfg_file "Sensor1_Entry.cfg" $CAM_NAME $DRV_FILE $MODE_FILE $MODE
+                         ;;
 		 *)
 			echo "ISP configuration \"$ISP_CONFIG\" unsupported."
 			echo -e "$USAGE" >&2
